KachnaOnline.Business/Mappings/EventsMappings.cs

```diff
@@ -12,11 +15,33 @@ public class EventMappings : Profile
     {
         public EventMappings()
         {
-            this.CreateMap<EventEntity, Event>().ReverseMap();
+            this.CreateMap<EventEntity, Event>()
+                .ForMember(e => e.LinkedPlannedStateIds, options =>
+                    options.MapFrom(e => e.LinkedPlannedStates.Select(state => state.Id).ToList()));
+            this.CreateMap<Event, EventEntity>();
+
+            this.CreateMap<EventEntity, EventWithLinkedStates>()
+                .ForMember(e => e.LinkedStates, options =>
+                    options.MapFrom(e => e.LinkedPlannedStates))
+                .ForMember(e => e.LinkedPlannedStateIds, options =>
+                    options.MapFrom(e => e.LinkedPlannedStates.Select(state => state.Id).ToList()));
+
             this.CreateMap<EventEntity, NewEvent>().ReverseMap();
-            this.CreateMap<EventEntity, ModifiedEvent>().ReverseMap();
+            this.CreateMap<EventEntity, ModifiedEvent>()
+                .ForMember(e => e.LinkedPlannedStateIds, options =>
+                    options.MapFrom(e => e.LinkedPlannedStates.Select(state => state.Id).ToList()));
+            this.CreateMap<ModifiedEvent, EventEntity>();
+
             this.CreateMap<Event, EventDto>().ReverseMap();
             this.CreateMap<Event, ManagerEventDto>().ReverseMap();
+
+            this.CreateMap<EventWithLinkedStates, EventWithLinkedStatesDto>()
+                .ForMember(e => e.LinkedStatesDtos, options =>
+                    options.MapFrom(e => e.LinkedStates));
+            this.CreateMap<EventWithLinkedStates, ManagerEventWithLinkedStatesDto>()
+                .ForMember(e => e.LinkedStatesDtos, options =>
+                    options.MapFrom(e => e.LinkedStates));
+
             this.CreateMap<NewEvent, BaseEventDto>().ReverseMap();
             this.CreateMap<ModifiedEvent, BaseEventDto>().ReverseMap();
```

Comment:
Why do you need reverse mappings for these?

Reply:
Never mind, I have gone through them and removed unnecessary mappings, where it was possible.